src/Discord.Net.SlashCommands/TypeConverters/DefaultEntityTypeConverter.cs

```diff
@@ -0,0 +1,37 @@
+using Discord.WebSocket;
+using System;
+using System.Threading.Tasks;
+
+namespace Discord.SlashCommands
+{
+    internal abstract class DefaultEntityTypeConverter<T> : TypeConverter<T> where T : class
+    {
+        public override Task<TypeConverterResult> ReadAsync (ISlashCommandContext context, SocketSlashCommandDataOption option, IServiceProvider services)
+        {
+            if (option.Value != null)
+                return Task.FromResult(TypeConverterResult.FromSuccess(option.Value as T));
```

Comment:
Dont think this works 🗡️

Reply:
It is easy enough to use a switch to assign parameters but the idea is to allow the users to use their own implementation for this. Using the same TypeConverter pattern makes the parameter assignments universal throughout the library.